### PR TITLE
handling exception read permission

### PR DESCRIPTION
**Pull request by eddynaka**
Fix Issue #657.
<Short description of the fix.>

- [X] I ran Unit Tests locally.
- [X] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue.

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

**Comment by eddynaka** [comment]
@cijothomas just created a new PR for this issue. Let me know if that is correct!
Thanks

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 1]
```diff
@@ -177,17 +178,25 @@ public IDebugOutput GetDebugOutput()
...
-            return !string.IsNullOrEmpty(value);
+            try
+            {
+                value = Environment.GetEnvironmentVariable(name);
+                return !string.IsNullOrEmpty(value);
+            }
+            catch (Exception e)
+            {
+                CoreEventSource.Log.FailedToLoadEnvironmentVariables(e.ToString());
+                throw;
```
Comment (TimothyMothra) [inline BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: a Try/Get method should not throw exceptions

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 2]
```diff
@@ -196,7 +205,15 @@ public bool TryGetEnvironmentVariable(string name, out string value)
...
         {
-            return Environment.GetEnvironmentVariable("COMPUTERNAME");
+            try
+            {
+                return Environment.GetEnvironmentVariable("COMPUTERNAME");
+            }
+            catch (Exception e)
+            {
+                CoreEventSource.Log.FailedToLoadEnvironmentVariables(e.ToString());
+                throw;
```
Comment (TimothyMothra) [inline BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: We don't want these methods to throw exceptions. The EventSource log will create internal traces that we can monitor.

@cijothomas what is the correct default value to return in this instance?

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 3]
```diff
@@ -33,9 +32,10 @@ public PlatformImplementation()
             {
                 this.environmentVariables = Environment.GetEnvironmentVariables();
             }
-            catch (SecurityException e)
+            catch (Exception e)
             {
                 CoreEventSource.Log.FailedToLoadEnvironmentVariables(e.ToString());
+                throw;
```
Comment (TimothyMothra) [inline BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: Catching the generic exception is correct :)

But we don't want these methods to throw exceptions. The EventSource log will create internal traces that we can monitor.

@cijothomas what is the correct default value to return in this instance?

**Comment by TimothyMothra** [comment]
Please update the Changelog under the VNext section

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 1 above)
Comment (eddynaka) [reply BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: ok. just removed!

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 2 above)
Comment (eddynaka) [reply BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: Just removed the throw just like the others

**Comment by TimothyMothra** [comment]
@cijothomas do you have time to re-review this one?

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 2 above)
Comment (snakefoot) [reply BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: But the `throw;` is still there.

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 4]
```diff
@@ -177,17 +177,24 @@ public IDebugOutput GetDebugOutput()
...
-            value = Environment.GetEnvironmentVariable(name);
-            return !string.IsNullOrEmpty(value);
+            try
+            {
+                value = Environment.GetEnvironmentVariable(name);
+                return !string.IsNullOrEmpty(value);
+            }
+            catch (Exception e)
+            {
+                CoreEventSource.Log.FailedToLoadEnvironmentVariables(e.ToString());
```
Comment (snakefoot) [inline BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: Missing `return false;` ?

File: BASE/Test/TestFramework/Shared/StubPlatform.cs [hunk 5]
```diff
@@ -23,8 +24,15 @@ public IDebugOutput GetDebugOutput()
...
-            value = Environment.GetEnvironmentVariable(name);
-            return !string.IsNullOrEmpty(value);
+            try
+            {
+                value = Environment.GetEnvironmentVariable(name);
+                return !string.IsNullOrEmpty(value);
+            }
+            catch (Exception e)
+            {
+                CoreEventSource.Log.FailedToLoadEnvironmentVariables(e.ToString());
```
Comment (snakefoot) [inline BASE/Test/TestFramework/Shared/StubPlatform.cs]: Missing `return false;` ?

Comment (eddynaka) [reply BASE/Test/TestFramework/Shared/StubPlatform.cs]: just pushed another commit returning false! Thanks!

File: BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 4 above)
Comment (eddynaka) [reply BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: just pushed another commit returning false! Thanks!

**Pull request merged**